src/Combinatorics/EnumerativeCombinatorics/combinations.jl

```diff
@@ -18,7 +18,7 @@ julia> collect(C)
  [2, 3, 4]
 ```
 """
-combinations(n::T, k::IntegerUnion) where T<:IntegerUnion = Combinations(Base.OneTo(n), n, T(k))
+combinations(n::T, k::IntegerUnion) where T<:IntegerUnion = Combinations(Base.oneto(n), n, T(k))
```

Comment:
Just to say, this also means a difference in behavior between this constructor, and the one taking an `AbstractVector`: here the type of `k` is ignored, the type of `n` alone decided what we do. There, the type of `k` decides which type is used for the "indices".

Compare:
```julia-repl
julia> typeof(combinations(3, Int8(2)))
Oscar.Combinations{Base.OneTo{Int64}, Int64}

julia> typeof(combinations(1:3, Int8(2)))
Oscar.Combinations{UnitRange{Int64}, Int8}
```
and
```julia-repl
julia> typeof(combinations(Int8(3), 2))
Oscar.Combinations{Base.OneTo{Int8}, Int8}

julia> typeof(combinations(Int8(1):Int8(3), 2))
Oscar.Combinations{UnitRange{Int8}, Int64}
```

This may be surprising to users. So perhaps we should restrict this again to
```suggestion
combinations(n::T, k::T) where T<:IntegerUnion = Combinations(Base.oneto(n), n, k)
```
That wouldn't be break as in Oscar 1.5.0 the method still is `combinations(n::Int, k::Int)`.

Reply:
hm, yeah I liked being able to do `combinations(Int8(3), 2)` because I'm lazy, but seeing all of this I agree that it is inconsistent / not user friendly.